sys/hashes/sha512_common.c

```diff
@@ -34,26 +34,25 @@
 #else /* !__BIG_ENDIAN__ */
 
 /*
- * Encode a length len/8 vector of (uint64_t) into a length len vector of
- * (unsigned char) in big-endian form.  Assumes len is a multiple of 8.
+ * Encode a length ceil(len/8) vector of (uint64_t) into a length len vector of
+ * (unsigned char) in big-endian form.
  */
 static void be64enc_vect(void *dst_, const void *src_, size_t len)
 {
-    /* Assert if len is not a multiple of 8 */
-    assert(!(len & 7));
```

Comment:
What is the reasoning behind this deletion?

Reply:
This change was necessary to implement SHA-512/224, since 224bit = 28B are not a multiple of 8. This requirement was imposed before to be sure all bytes are copied in the unrolled copy operation.

The potential left-over bytes (`i < len`) are now copied on demand below inside the if case.